### PR TITLE
Prp-220 Changes page - rendering, form repopulation, validation

### PR DESCRIPTION
**Pull request by microwavenby**
## Ticket

https://wicmtdp.atlassian.net/browse/PRP-220

## Changes
- Changes page created
- Form validator requires answers to both Name/ID changes and Moved in the past year
- Form Action upserts a SubmissionPage for the participant's submissionID containing the information in the form
- Form Action routes to the `/upload` page if yes was answered to any question on this page
- Storybook updated
- Unit tests for page validator
- Unit tests for page routing
- e2e tests for form submission and redirection to upload

Screenshot 2023-04-04 at 8 35 27 PM

### CODE REVIEW

**Comment by makaelastephens** [comment]
Sorry, we didn't include this in the LucidChart, but we need to add the RequiredQuestionStatement and mark both questions as required. Updating the LC...

File: bin/run-e2e.sh [hunk 1]
```diff
@@ -3,6 +3,7 @@
 set -euo pipefail
 
 COMMAND=$1
+EXTRA_ARG=${2:-""}
```
Comment (rocketnova) [inline bin/run-e2e.sh:6]: What is this for? 👀

File: participant/package.json [hunk 2]
```diff
@@ -55,7 +56,7 @@
     "react-number-format": "^5.1.2",
     "remix-i18next": "^4.2.0",
     "remix-image": "^1.3.3",
-    "remix-validated-form": "^4.6.5",
+    "remix-validated-form": "4.6.5",
```
Comment (rocketnova) [inline participant/package.json:59]: Why did we need to pin?

File: participant/app/components/List.tsx [hunk 3]
```diff
@@ -1,32 +1,35 @@
...
   const ListTag = type == "ordered" ? "ol" : "ul";
 
   return (
     <ListTag className={classNames.trim()}>
-      {listKeys.map((key: string) => (
-        <li key={key}>
-          <Trans i18nKey={key} />
-        </li>
+      {listItems.map((item: string) => (
+        <li key={item}>{item}</li>
```
Comment (rocketnova) [inline participant/app/components/List.tsx]: You can actually run the string through the `<Trans>` component here to retain the ability to add simple html tags to the i18n strings.

```suggestion
        <li key={item}><Trans i18nKey={item} /></li>
```

File: participant/app/utils/validation.ts [hunk 4]
```diff
@@ -20,3 +17,14 @@ const nameSchemaFactory = (idPrefix: string) => {
...
 export const representativeNameSchema = nameSchemaFactory("representative");
+
+export const changesSchema = zfd.formData({
+  idChange: zfd.text(z.enum(["yes", "no"])),
+  addressChange: zfd.text(z.enum(["yes", "no"])),
+});
+
+export const countSchema = zfd.formData({
+  householdSize: zfd.numeric(
+    z.number().min(1, { message: "You must recertify for at least one person" })
```
Comment (rocketnova) [inline participant/app/utils/validation.ts:28]: I think I asked this previously and forgotten the answer: Can this be an i18n string?

File: participant/app/types.ts [hunk 5]
```diff
@@ -1,3 +1,14 @@
...
+export type RouteType = "changes";
+
+export type ChangesData = {
+  idChange: string;
+  addressChange: string;
+};
+
+export type HouseholdSizeData = {
+  householdSize: number;
+};
```
Comment (rocketnova) [inline participant/app/types.ts]: What is this one for?

File: participant/e2e/routes/changes.spec.ts [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+  expect(parsedPostData).toMatchObject({
+    __rvfInternalFormId: "changesForm",
+    idChange: "yes",
+    addressChange: "yes",
+  });
+
+  // Check that we've moved to the Upload page based on our changes
+  await expect(page).toHaveURL("/gallatin/recertify/upload");
+
+  // Capture the GET request when we load the /eligibility page again by clicking "Back"
```
Comment (rocketnova) [inline participant/e2e/routes/changes.spec.ts]:
```suggestion
  // Capture the GET request when we load the /changes page again by clicking "Back"
```

**Review by rocketnova — COMMENTED** [review]
Some questions.

File: participant/app/types.ts (hunk 5 above)
Comment (microwavenby) [reply participant/app/types.ts]: I definitely did a bunch of casting from form pages back to types explicitly for repopulation of forms. Guess this was preemptive? 🤷🏻‍♀️

File: participant/app/utils/validation.ts (hunk 4 above)
Comment (microwavenby) [reply participant/app/utils/validation.ts:28]: I've no idea. We can try it.. in a ticket that isn't about the changes page? 🤔

File: participant/app/types.ts (hunk 5 above)
Comment (microwavenby) [reply participant/app/types.ts]: (removing it)

File: participant/package.json (hunk 2 above)
Comment (microwavenby) [reply participant/package.json:59]: This was the change that fixed my ability to mock `useField` in jest that i spent 3+ hours fighting with yesterday.

File: bin/run-e2e.sh (hunk 1 above)
Comment (microwavenby) [reply bin/run-e2e.sh:6]: So you can run e2e with a single spec file
`npm run e2e e2e/routes/changes.spec.ts`

Comment (rocketnova) [reply bin/run-e2e.sh:6]: Oh got it!

File: participant/package.json (hunk 2 above)
Comment (rocketnova) [reply participant/package.json:59]: 😞 oh jest

**Review by rocketnova — APPROVED** [review]
LGTM! 🚢 Thank you!

**Pull request merged**